core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -370,6 +424,11 @@ else if (partitioning.equals(SCALED_WRITER_DISTRIBUTION)) {
             bufferCount = defaultConcurrency;
             checkArgument(partitionChannels.isEmpty(), "Scaled writer exchange must not have partition channels");
         }
+        else if (partitioning.equals(SCALED_WRITER_HASH_DISTRIBUTION) || isScalingPartitioning(partitioning)) {
+            // Even when scale writers is enabled, the buffer count or the number of drivers will remain constant.
+            // However, only some of them are actively doing the work.
```

Comment:
`However, only some of them are actively doing the work.` -> `However, only some of them might be actively doing the work.`